_test/general.test.php

```diff
@@ -58,4 +78,408 @@ public function test_plugin_conf() {
         }
 
     }
+
+    //======================================================================
+    // XML TESTS
+    //======================================================================
+
+    /**
+     * Test to ensure that XML is valid when DokuWiki has no pages.
+     */
+    public function test_xml_response_is_valid_if_dokuwiki_has_no_pages() {
+        $conf = array();
+        $DokuwikiXMLExport = new DokuwikiXMLExport($conf);
+        $xml = new SimpleXMLElement($DokuwikiXMLExport->generateXMLExport(0, 20));
+
+        $start = implode('', ($xml->xpath('/findologic/items/@start')));
+        $count = implode('', ($xml->xpath('/findologic/items/@count')));
+        $total = implode('', ($xml->xpath('/findologic/items/@total')));
+
+        $expected = 0;
+        $this->assertEquals($expected, $start, 'Expected start value should match "0" when DokuWiki has no pages.');
+        $this->assertEquals($expected, $count, 'Expected count value should match "0" when DokuWiki has no pages.');
```

Comment:
I believe `count` should always match the value from the request, so in this case `20`.

Reply:
Yes but there are two cases in the code where i explicitly check this value and change it accordingly.

In `/DokuwikiXMLExport.php`:
`$count = min($total, $count);`
and afterwards:

```
if (($count + $start) > $total) {
    throw new \InvalidArgumentException("Error: Failed while trying to validate start and count values");
}
```

This would require at least two code changes.
@georgms should I make an issue?

Reply:
Never change `count`, it should always reflect the value from the request. I suggest you fix it in this PR so you don't break tests.

Reply:
@georgms the library `libflexport` sets the count value according to the products that have been transferred. I do not think that there is a solution?

Reply:
I see. I've added https://github.com/findologic/libflexport/issues/18 to fix libflexport. The logic in `DokuwikiXMLExport.php` should still be changed, as far as I can tell this should not affect the generated XML.